### ISSUE TITLE
[Esame Gennaio 2022] Esercizio 1

### ISSUE THREAD

**Issue by matypist** [sapienzastudentsnetwork/psd2223#83]
**Esercizio 1 (8 punti)**

Progettare un circuito sequenziale con un ingresso x e due uscite z1 e z0. L’uscita z1 deve essere uguale a 1 se gli ultimi tre bit di ingresso contengono esattamente due 1, mentre z0 deve essere 1 se gli ultimi 2 bit sono uguali. Si utilizzino FF di tipo SR per il bit di stato più significativo e FF di tipo T per i restanti bit. Non si considerino le sovrapposizioni.

Esempio
```
x  0101110010000011
z1 0001000000000001
z0 0000010100101001
```

**Comment by Elia-Belli** [comment]
![1](https://user-images.githubusercontent.com/107944476/211845169-2a8aba76-6dd8-419a-aab4-d3d03cf1bf60.jpg)

**Comment by GrenzGaenger450** [comment]
il circuito mi sembra sbagliato, mancano gli s? e s00 s11.
Infatti nell'esame 2023 ci sono 7 stati e non 5

**Comment by Elia-Belli** [comment]
> il circuito mi sembra sbagliato, mancano gli s? e s00 s11. Infatti nell'esame 2023 ci sono 7 stati e non 5

C'è una piccola differenza rispetto all'esame del 18/01/2023 e per accorgersene bisogna leggere attentamente le condizioni degli output e osservare gli esempi:

- In questo caso nell'esempio notiamo che non appena si ricevono 2 bit uguali l'output $z_0$ diventa 1 senza aspettare il terzo bit perchè la condizione è "se gli ultimi 2 bit sono uguali", di conseguenza gli stati $S00$ e $S11$ sono inutili perchè si sà già che, non dovendo considerare le sovrapposizioni, si tornerà in $R$.
```
x  0101110010000011
z1 0001000000000001
z0 0000010100101001
```
- Nell'esame del 18/01/2023 invece entrambe le condizioni si basavano su sequenze di 3 bit, quindi lo stato $S00$ era necessario perchè nonostante sapessimo già che $z_1$ sarebbe stato 1 bisognava aspettare il terzo bit della sequenza e controllare anche $z_0$

Allego il testo dell'esame:
> Progettare un circuito sequenziale con un ingresso x due uscite z1 e z0. L’uscita z1 deve essere uguale
a 1 se gli ultimi tre bit di ingresso contengono almeno due 0, mentre z0 deve essere 1 se gli ultimi 3
bit sono uguali. Non si considerino le sovrapposizioni.
```
x  10100000111
z1 00010010000
z0 00000010001
```

**Comment by Elia-Belli** [comment]
> quindi in z1 dovrei considerare solo 2 input invece di 3?

No, è l'uscita $z_0$ a dipendere da sequenze di 2 bit, mentre $z_1$ da sequenze di 3.

**Comment by fraxb04** [comment]
@Elia-Belli 
brodomo credo ci sia un errore nell'encoding di R = 111, che in teoria seguendo la logica dovrebbe essere 100. 
Tuttavia non credo cambi il risultato

**Comment by Elia-Belli** [comment]
> @Elia-Belli brodomo credo ci sia un errore nell'encoding di R = 111, che in teoria seguendo la logica dovrebbe essere 100. Inoltre non mi è chiaro se l'esercizio richieda di fare anche il disegno del circuito

Non è un errore, mi avanzano le codifiche $111$, $100$, $101$ e $110$ e ne ho scelta una a caso per $R$, non sono assegnate secondo alcuna logica le codifiche. 
Comunque si, l'esercizio richiederebbe il circuito ma trovate le equazioni, minimizzarle e disegnarlo è banale, volendo potete ripartire da dove ho finito l'esercizio ed aggiungere voi il circuito.

**Comment by fraxb04** [comment]
no, poiché se scegliessi 100 al posto di 111, verrebbero delle equazioni minimizzate in modo diverso

**Comment by Elia-Belli** [comment]
Ma è normale, se hai fatto le codifiche in modo diverso è normale che le equazioni vengano diverse, ma lo stato a cui si fa riferimento è sempre lo stesso

**Comment by fraxb04** [comment]
@Elia-Belli 
certamente, lo stato è lo stesso
